src/app/modules/main/wallet_section/saved_addresses/model.nim

```diff
@@ -107,25 +102,17 @@ QtObject:
     for item in items:
         self.itemChanged(item.getAddress())
 
-  proc getItemByAddress*(self: Model, address: string): Item =
+  proc getItemByAddress*(self: Model, address: string, isTest: bool): Item =
     if address.len == 0 or address == ZERO_ADDRESS:
       return
     for item in self.items:
-      if cmpIgnoreCase(item.getAddress(), address) == 0:
-        return item
-
-  proc getItemByEnsOrAddress*(self: Model, addrOrEns: string): Item =
-    if addrOrEns.len == 0:
-      return
-    for item in self.items:
-      if item.getEns().len > 0:
-        if item.getEns() == addrOrEns:
+      if cmpIgnoreCase(item.getAddress(), address) == 0 and
+        (item.getIsTest() and isTest or not item.getIsTest() and not isTest):
```

Comment:
```suggestion
        (item.getIsTest() == isTest):
```

Reply:
I wrote it firstly like that, but then changed it cause I thought one who looks at it would understand the condition easier.

Now, not sure what's easier for reading since you suggested that.

Reply:
will update it